gem/gem.py

```diff
@@ -213,6 +265,25 @@ def __new__(cls, a, b):
         self.children = a, b
         return self
 
+    def latex(self):
+        a, b = self.children
+        return r'({0} + {1})'.format(a.latex(), b.latex())
+
+    def _repr_latex_(self):
+        return r'${0}$'.format(self.latex())
+
+    def is_equal(self, other):
+        a, b = self.children
+        c, d = other.children
+        return (a == c and b == d) or (a == d and b == c)
+
+    def get_hash(self):
+        a, b = self.children
+        if hash(a) < hash(b):
+            return hash((Sum, a, b))
+        else:
+            return hash((Sum, b, a))
```

Comment:
The practical reason is that I want to factor out (a+b) from `(a+b)*c + (b+a)*d`. Not sure how often that will help though. For products, I'm factor out each factor individually (`a`, then `b`, not `a*b` together), so factorisation should proceed anyway.
Will implement `CommutativeOperator` next.